### PR TITLE
Add scripts for testing transpilation against mps

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This adds some small scripts for testing the success rate on mozilla-pipeline-schemas. There are currently 25 schemas that fail to convert at all. I think a significant portion of these can be attributed to things like mozilla-services/mozilla-pipeline-schemas#288. 

Here's an example of the output:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//pocket/fire-tv-events/fire-tv-events.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//org-mozilla-reference-browser/metrics/metrics.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//org-mozilla-reference-browser/baseline/baseline.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//org-mozilla-reference-browser/events/events.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//org-mozilla-samples-glean/metrics/metrics.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//org-mozilla-samples-glean/baseline/baseline.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//org-mozilla-samples-glean/events/events.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("data did not match any variant of untagged enum AdditionalProperties", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//firefox-launcher-process/launcher-process-failure/launcher-process-failure.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//glean/metrics/metrics.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//glean/baseline/baseline.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//glean/events/events.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/mobile-event/mobile-event.1.schema.json
thread 'main' panicked at 'array missing item', src/jsonschema.rs:191:21
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/testpilot/testpilot.4.schema.json
thread 'main' panicked at 'array missing item', src/jsonschema.rs:191:21
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/update/update.4.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/untrustedModules/untrustedModules.4.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/focus-event/focus-event.1.schema.json
thread 'main' panicked at 'array missing item', src/jsonschema.rs:191:21
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/crash/crash.4.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/shield-icq-v1/shield-icq-v1.4.schema.json
thread 'main' panicked at 'array missing item', src/jsonschema.rs:191:21
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/new-profile/new-profile.4.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/main/main.4.schema.json
thread 'main' panicked at 'array missing item', src/jsonschema.rs:191:21
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/modules/modules.4.schema.json
thread 'main' panicked at 'array missing item', src/jsonschema.rs:191:21
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/mobile-metrics/mobile-metrics.1.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/event/event.4.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//telemetry/saved-session/saved-session.4.schema.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//mozza/event/event.1.schema.json
25/65 failures
```

### CODE REVIEW

**Comment by acmiyaguchi** [comment]
The new script that I've added found a pretty serious bug with nested records. The logs can be found here:

https://gist.github.com/acmiyaguchi/670655fdc86b65bd2bc56923b3be4456

**Comment by badboy** [comment]
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: sequence, expected struct Tag", line: 0, column: 0)', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Failed on schemas//pocket/fire-tv-events/fire-tv-events.1.schema.json
```

He, I told you: proper error handling will get you nicer error messages in the end as well :P

File: scripts/test-mps-valid-avro.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    return schema
+
+
+def parse_schema_name(path):
+    """Given a directory path to a json schema in the mps directory, generate
+    the fully qualified name in the form `{namespace}.{doctype}.{docver}`."""
+    elements = path.split("/")
+    doctype, docver = elements[-1].split(".")[:-2]
+    namespace = elements[-3]
+    return f"{namespace}.{doctype}.{docver}"
```
Comment (badboy) [inline scripts/test-mps-valid-avro.py]: Python noob here, but curious: What does the `f` prefix do?

Comment (acmiyaguchi) [reply scripts/test-mps-valid-avro.py]: It's the python 3 syntax for literal string interpolation ([pep 498](https://www.python.org/dev/peps/pep-0498/)). It's clean.

**Pull request merged**